code/pages/SubsitesVirtualPage.php

```diff
@@ -170,24 +196,44 @@ public function onBeforeWrite()
             $this->ExtraMeta = $this->ContentSource()->ExtraMeta ? $this->ContentSource()->ExtraMeta : $this->ExtraMeta;
         }
     }
-}
 
-class SubsitesVirtualPage_Controller extends VirtualPage_Controller
-{
-    public function reloadContent()
+    public function validURLSegment()
     {
-        $this->failover->copyFrom($this->failover->CopyContentFrom());
-        $this->failover->write();
-        return;
-    }
+        $isValid = parent::validURLSegment();
+
+        // Veto the validation rules if its false. In this case, some logic
+        // needs to be duplicated from parent to find out the exact reason the validation failed.
+        if (!$isValid) {
+            $IDFilter = $this->ID ? "AND \"SiteTree\".\"ID\" <> $this->ID" : null;
```

Comment:
Please parameterise these queries rather than `" <> $this->ID"` variable substitution.